### PR TITLE
Format files to prepare for pypi publish

### PR DESCRIPTION
**Pull request by ajlee21**
This PR updates Readme and setup files to prepare for publishing this repo to pypi

### CODE REVIEW

**Comment by ajlee21** [comment]
The linter error is due to being on a fork: https://github.com/samuelmeuli/lint-action/issues/13

The analysis error is because random is a requirement that is not found in pypi anymore. Its not called random2. I will change the code to use numpy random in this case.

**Comment by ajlee21** [comment]
Based on discussion with @cgreene pulling out "extra" functions (i.e. functions specific to analysis performed in https://github.com/greenelab/simulate-expression-compendia) and leave only simulation functions (i.e dir setup, vae training, simulations).

Closing it for now while I add those changes and will reopen when its ready for review

**Comment by ajlee21** [comment]
This PR is now ready to review. 

This PR complements: https://github.com/greenelab/simulate-expression-compendia/pull/33

The ponyo repo included scripts to about the method to generate new simulated data as well as scripts specific to running the analysis found in this repo. This PR is removes those analysis-specific scripts.

The main changes in this PR are:
1. Renaming and moving scripts in `ponyo/`
2. Adding test notebooks to test each type of simulation approach `human_tests`
3. Update Readme and setup.py files to get ready for pypi deployment.

File: ponyo/utils.py [hunk 1]
```diff
@@ -11,3 +16,154 @@ def read_config(filename):
...
+    train_architecture = params["NN_architecture"]
+
+    # Create VAE directories
+    output_dirs = [
+        os.path.join(base_dir, dataset_name, "models"),
+        os.path.join(base_dir, dataset_name, "logs"),
+    ]
+
+    # Check if analysis output directory exist otherwise create
+    for each_dir in output_dirs:
```
Comment (dongbohu) [inline ponyo/utils.py:49]: Does it make more sense to merge this `for` loop with the one on lines 55-59?

File: ponyo/utils.py [hunk 2]
```diff
@@ -11,3 +16,154 @@ def read_config(filename):
...
+
+        # Check if NN architecture directory exist otherwise create
+    for each_dir in output_dirs:
+        new_dir = os.path.join(each_dir, train_architecture)
+        if not os.path.exists(new_dir):
+            print("creating new directory: {}".format(new_dir))
+            os.makedirs(new_dir, exist_ok=True)
+
+    # Create results directories
+    output_dirs = [os.path.join(base_dir, dataset_name, "results")]
```
Comment (dongbohu) [inline ponyo/utils.py:62]: This list includes only one element. The `for` loops on lines 65 and 71 seem redundant. If you do want to use `for` loop, these two loops can be merged into one.

File: README.md [hunk 3]
```diff
@@ -16,6 +16,31 @@ pip install ponyo
...
+The tables lists the core parameters required to generate simulated data using modules from [ponyo](https://github.com/greenelab/ponyo). Those marked with * indicate those parameters that will vary depending on the type of approach .
+
+| Name | Description |
+| :--- | :---------- |
+| local_dir| str: Parent directory on local machine to store intermediate results|
+| scaler_transform_file| str: Files to store mapping from normalized to raw gene expression range|
+| dataset_name| str: Name for analysis directory. Either "Human" or "Pseudomonas"|
+| simulation_type | str: "sample_lvl_sim" or "experiment_lvl_sim"|
+| NN_architecture | str: Name of neural network architecture to use. Format 'NN_<intermediate layer>_<latent layer>'|
+| learning_rate| float: Parent directory on local machine to store intermediate results|
```
[the remark this answers is not in the document]
Comment (ajlee21) [reply README.md:34]: Good catch! Found additional copy-paste errors too

File: .github/workflows/tests.yaml [hunk 4]
```diff
@@ -20,16 +20,13 @@ jobs:
         run: |
           # $CONDA is an environment variable pointing to the root of the miniconda directory
           $CONDA/bin/conda env update --file environment.yml --name base
-          $CONDA/bin/pip install .
+          $CONDA/bin/pip install -e .
```
Comment (ben-heil) [inline .github/workflows/tests.yaml:23]: Minor note, you don't need the installation to be editable because presumably you're not modifying the code with your github actions

File: environment.yml [hunk 5]
```diff
@@ -1,12 +1,7 @@
-name: simulate_expression_compendia
+name: test_ponyo
```
Comment (ben-heil) [inline environment.yml:1]: Why is your environment called `test_ponyo` instead of `ponyo`?

Also if this package is installable with pip why have a `environment.yml` file instead of a `requirements.txt` file?

File: ponyo/helper_vae.py [hunk 6]
```diff
@@ -32,7 +31,7 @@
 # The below is necessary for starting core Python generated random numbers
 # in a well-defined state.
 
-rn.seed(12345)
+np.random.seed(12345)
```
Comment (ben-heil) [inline ponyo/helper_vae.py:34]: I saw one of your earlier comments that said that random wasn't a package available in pypi. Shouldn't random be built into the python language? Is `random2` a package designed to overwrite the normal behavior of the random package?

File: ponyo/vae.py [hunk 7]
```diff
@@ -106,7 +105,7 @@ def tybalt_2layer_model(
     # The below is necessary for starting core Python generated random numbers
     # in a well-defined state.
 
-    rn.seed(12345)
+    np.random.seed(12345)
```
Comment (ben-heil) [inline ponyo/vae.py:108]: You're setting the numpy seed twice in this file, and the seed for the base `random` package is no longer set. The second thing may not matter since you're not importing random in this file, but the first one appears to be a copy number variation

File: setup.cfg [hunk 8]
```diff
@@ -7,4 +7,5 @@ exclude =
     .git,
     setup.py,
     ponyo/cca_core.py
```
Comment (ben-heil) [inline setup.cfg:9]: You can omit this line since `cca_core` doesn't exist anymore

File: setup.py [hunk 9]
```diff
@@ -1,13 +1,23 @@
+import pathlib
 from setuptools import setup
 
+# The directory containing this file
+HERE = pathlib.Path(__file__).parent
+
+# The text of the README file
+README = (HERE / "README.md").read_text()
```
Comment (ben-heil) [inline setup.py:8]: That's cool, I didn't know that you could use a `/` like that in pathlib

File: ponyo/helper_vae.py (hunk 6 above)
Comment (ajlee21) [reply ponyo/helper_vae.py:34]: I believe you're right about random being pre-packaged with python. I wasn't aware of that at the time when I was making the list of required packages so I ran into an issue when `random` wasn't found in pypi.

Comment (cgreene) [reply ponyo/helper_vae.py:34]: Are you sure that np.random.seed and random.seed do the same thing? In the very distant past (like, 10+ years ago), I think that the random number generators were not shared between python and numpy. You would have to set both (or just the one you were using) to get the desired behavior. Are you observing that the RNG behaves as you wish with either set, and that not setting one or the other produces undesired behavior?

Comment (ajlee21) [reply ponyo/helper_vae.py:34]: You are correct. This is a case of human error as I was refactoring imports. There should be two distinct random seeds set using random and numpy libraries. 

As discussed, I will also try to add a pytest to test reproducible outputs and see how that goes

**Comment by ajlee21** [comment]
Thank you for your comments!

@ben-heil about the linter, since the linter is working properly and its a permissions error with my fork I'll wait to see if this changes (since submitting feedback to them)
